dotCMS/src/main/java/com/dotcms/rest/ContentResource.java

```diff
@@ -1257,4 +1239,38 @@ private void setRequestMetadata(Contentlet contentlet, HttpServletRequest reques
 			
 		}
 	}
+
+	private Map<Relationship, List<Contentlet>> addContentRelationships(Contentlet contentlet, Structure st, Map<String,Object> map) 
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)